src/Form/Form.js

```diff
@@ -895,11 +895,24 @@ class Form {
 
         // After autofill we check if form values match the data provided…
         const formValues = this.getValuesReadyForStorage();
+        const hasNoCredentialsData = !formValues.credentials?.username && !formValues.credentials?.password;
+        const hasOnlyEmail =
+            formValues.identities && Object.keys(formValues.identities ?? {}).length === 1 && formValues.identities?.emailAddress;
+
+        const hasOnlyOneCredentialOrEmail =
+            Boolean(formValues.credentials?.username) !== Boolean(formValues.credentials?.password) ||
+            (hasOnlyEmail && hasNoCredentialsData);
         const areAllFormValuesKnown = Object.keys(formValues[dataType] || {}).every(
             (subtype) => formValues[dataType][subtype] === data[subtype],
         );
-        if (areAllFormValuesKnown) {
-            // …if we know all the values do not prompt to store data
+
+        // If we only have a single credential field - then we want to prompt a partial save with username,
+        // So that in multi step forms (like reset-password), we can identify which username was picked, or complete a password save.
+        if (hasOnlyOneCredentialOrEmail) {
+            this.shouldPromptToStoreData = true;
+            this.shouldAutoSubmit = this.device.globalConfig.isMobileApp;
```

Comment:
We don't want to autosubmit here, do we? Imagine people just generating a private duck address on a newsletter form. We don't want to automatically submit that. Am I missing anything?